tests/csapi/admin_test.go

```diff
@@ -13,3 +21,111 @@ func TestCanRegisterAdmin(t *testing.T) {
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: http.StatusForbidden,
+			JSON: []match.JSON{
+				match.JSONKeyEqual("errcode", "M_FORBIDDEN"),
+			},
+		})
+	})
+	t.Run("/send_server_notice as an admin is allowed", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+
+	})
+	t.Run("Alice is invited to the server alert room", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+		syncUntilInvite(t, alice)
+	})
+	t.Run("Alice cannot reject the invite", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+		roomID := syncUntilInvite(t, alice)
+		res := alice.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "rooms", roomID, "leave"})
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: http.StatusForbidden,
+			JSON: []match.JSON{
+				match.JSONKeyEqual("errcode", "M_CANNOT_LEAVE_SERVER_NOTICE_ROOM"),
+			},
+		})
+	})
+	t.Run("Alice can join the alert room", func(t *testing.T) {
+		eventID := sendServerNotice(t, admin, reqBody, nil)
+		roomID := syncUntilInvite(t, alice)
+		alice.JoinRoom(t, roomID, []string{})
+		// cleanup room, so the next test is in a clean state
+		defer alice.LeaveRoom(t, roomID)
+		queryParams := url.Values{}
+		queryParams.Set("dir", "b")
+		// check if we received the message
+		res := alice.DoFunc(t, "GET", []string{"_matrix", "client", "r0", "rooms", roomID, "messages"}, client.WithQueries(queryParams))
+		msgRes := &msgResult{}
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: http.StatusOK,
+			JSON: []match.JSON{
+				findMessageId(eventID, msgRes),
+			},
+		})
+		if !msgRes.found {
+			t.Errorf("did not find expected message from server notices")
+		}
+	})
+	t.Run("Alice can leave the alert room, after joining it", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+		roomID := syncUntilInvite(t, alice)
+		alice.JoinRoom(t, roomID, []string{})
+		alice.LeaveRoom(t, roomID)
+	})
+	t.Run("Sending a notice with a transactionID is idempotent", func(t *testing.T) {
+		txnID := "1"
+		eventID1 := sendServerNotice(t, admin, reqBody, &txnID)
+		eventID2 := sendServerNotice(t, admin, reqBody, &txnID)
+		if eventID1 != eventID2 {
+			t.Errorf("expected event IDs to be the same, but got '%s' and '%s'", eventID1, eventID2)
+		}
+	})
+
+}
+
+func sendServerNotice(t *testing.T, admin *client.CSAPI, reqBody client.RequestOpt, txnID *string) (eventID string) {
+	var res *http.Response
+	if txnID != nil {
+		res = admin.MustDoFunc(t, "PUT", []string{"_synapse", "admin", "v1", "send_server_notice", *txnID}, reqBody)
+	} else {
+		res = admin.MustDoFunc(t, "POST", []string{"_synapse", "admin", "v1", "send_server_notice"}, reqBody)
+	}
+	body := must.MatchResponse(t, res, match.HTTPResponse{
+		StatusCode: http.StatusOK,
+		JSON: []match.JSON{
+			match.JSONKeyPresent("event_id"),
+		},
+	})
+	return gjson.GetBytes(body, "event_id").Str
+}
+
+func syncUntilInvite(t *testing.T, alice *client.CSAPI) string {
```

Comment:
Needs a comment explaining why you can't use `client.SyncInvitedTo` - you don't know the room ID!